### PR TITLE
fix: updates executable response spec for executable-sourced credentials

### PR DESCRIPTION
**Pull request by lsirac**
For successful responses, the `expiration_time` field is only required
when an output file is specified in the credential configuration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -421,6 +421,9 @@ A sample executable error response:
 These are all required fields for an error response. The code and message
 fields will be used by the library as part of the thrown exception.
 
+For successful responses, the `expiration_time` field is only required
+when an output file is specified in the credential configuration. 
+
 Response format fields summary:
   * `version`: The version of the JSON output. Currently only version 1 is supported.
   * `success`: The status of the response. When true, the response must contain the 3rd party token,
```
Comment (aeitzman) [inline README.md:429]: the summary for the success field here still says the response must contain expiration if successful, change to say that expiration time is optional

File: README.md [hunk 2]
```diff
@@ -421,6 +421,9 @@ A sample executable error response:
 These are all required fields for an error response. The code and message
 fields will be used by the library as part of the thrown exception.
 
+For successful responses, the `expiration_time` field is only required
```
Comment (aeitzman) [inline README.md:424]: Add this to the comment with samples in PluggableAuthCredentials

**Review by aeitzman — APPROVED** [review]

File: oauth2_http/javatests/com/google/auth/oauth2/PluggableAuthHandlerTest.java [hunk 3]
```diff
@@ -218,6 +220,216 @@ void retrieveTokenFromExecutable_errorResponse_throws() throws InterruptedExcept
...
+      assertEquals(
+          "Error code INVALID_EXECUTABLE_RESPONSE: The executable response must contain the "
+              + "`expiration_time` field for successful responses when an output_file has been specified in the"
+              + " configuration.",
+          exception.getMessage());
+
+      // Validate executable not invoked.
+      verify(mockProcess, times(0)).destroyForcibly();
+      verify(mockProcess, times(0))
+          .waitFor(eq(Long.valueOf(options.getExecutableTimeoutMs())), eq(TimeUnit.MILLISECONDS));
```
Comment (clundin25) [inline oauth2_http/javatests/com/google/auth/oauth2/PluggableAuthHandlerTest.java:429]: Does this `waitFor` statement mock time or does it have a physical sleep?

File: oauth2_http/java/com/google/auth/oauth2/PluggableAuthHandler.java [hunk 4]
```diff
@@ -112,6 +112,18 @@ public String retrieveTokenFromExecutable(ExecutableOptions options) throws IOEx
       executableResponse = getExecutableResponse(options);
     }
 
+    // If an output file is specified, successful responses must contain the `expiration_time`
+    // field.
+    if (options.getOutputFilePath() != null
```
Comment (clundin25) [inline oauth2_http/java/com/google/auth/oauth2/PluggableAuthHandler.java:117]: nit: This is a lot of conditions in this statement. IMO could be broken up into variables or a helper function

**Review by clundin25 — APPROVED** [review]

**Review by TimurSadykov — CHANGES_REQUESTED** [review]
review in progress

File: oauth2_http/java/com/google/auth/oauth2/PluggableAuthHandler.java (hunk 4 above)
Comment (lsirac) [reply oauth2_http/java/com/google/auth/oauth2/PluggableAuthHandler.java:117]: ack

File: oauth2_http/javatests/com/google/auth/oauth2/PluggableAuthHandlerTest.java (hunk 3 above)
Comment (lsirac) [reply oauth2_http/javatests/com/google/auth/oauth2/PluggableAuthHandlerTest.java:429]: This actually just verifies that waitFor wasn't called - waitFor was not mocked to do anything in this test. When it is mocked, it returns right away, but we validate that it was called with the right timeout.

**Review by TimurSadykov — APPROVED** [review]
LGTM

**Pull request merged**